crates/api/src/lib.rs

```diff
@@ -563,31 +563,291 @@ where
+                    namespace,
+                    id,
+                    activity,
+                }) => {
+                    model.namespace_context(namespace);
+                    let model = self.store.apply_prov_model_for_activity_id(
+                        connection,
+                        model,
+                        activity,
+                        namespace.external_id_part(),
+                    )?;
+
+                    self.store.apply_prov_model_for_entity_id(
+                        connection,
+                        model,
+                        id,
+                        namespace.external_id_part(),
+                    )?
+                }
+                ChronicleOperation::EntityDerive(EntityDerive {
+                    namespace,
+                    id,
+                    used_id,
+                    activity_id,
+                    ..
+                }) => {
+                    model.namespace_context(namespace);
+                    let model = self.store.apply_prov_model_for_entity_id(
+                        connection,
+                        model,
+                        id,
+                        namespace.external_id_part(),
+                    )?;
+
+                    let model = self.store.apply_prov_model_for_entity_id(
+                        connection,
+                        model,
+                        used_id,
+                        namespace.external_id_part(),
+                    )?;
+
+                    if let Some(id) = activity_id {
+                        self.store.apply_prov_model_for_activity_id(
+                            connection,
+                            model,
+                            id,
+                            namespace.external_id_part(),
+                        )?
+                    } else {
+                        model
+                    }
+                }
+                ChronicleOperation::WasAttributedTo(WasAttributedTo {
+                    namespace,
+                    entity_id,
+                    agent_id,
+                    ..
+                }) => {
+                    model.namespace_context(namespace);
+                    let model = self.store.apply_prov_model_for_entity_id(
+                        connection,
+                        model,
+                        entity_id,
+                        namespace.external_id_part(),
+                    )?;
+
+                    self.store.apply_prov_model_for_agent_id(
+                        connection,
+                        model,
+                        agent_id,
+                        namespace.external_id_part(),
+                    )?
+                }
+            };
+            let state = applied_model.clone();
+            applied_model.apply(op)?;
+            if state != applied_model {
+                info!("Compare: {:#?} != {:#?}", state, applied_model);
```

Comment:
Probably push this up to debug or trace?

Reply:
Leftovers from debugging. Thanks